### ISSUE TITLE
Plugin not displaying

### ISSUE THREAD

**Issue by adam-karnet** [sarhugo/strapi-plugin-cloudflare-pages#2]
Hi, your plugin is something that I would very much appreciate to use, however with **Strapi 4.2.2** (latest) and your version **0.4.0** when plugin is installed according to README.md ... nothing happens. It does not get displayed in the left menu in the application (at least not in develop mode). I can see the instructions are the same for v3 and v4 ... is it possible something else needs to be changed in version 4?

**Comment by sarhugo** [comment]
Hi, did you add the configuration at config/plugins.js and rebuild the admin?

**Comment by adam-karnet** [comment]
Yes, here is the content of **config/plugins.js** - 

```
module.exports = ({ env }) => ({
  'cloudflare-pages': {
    instances: [
      {
        name: "Production build",
        hook_url: 'https://api.cloudflare.com/client/v4/pages/webhooks/deploy_hooks/...'
      },
      {
        name: "Staging build",
        hook_url: 'https://api.cloudflare.com/client/v4/pages/webhooks/deploy_hooks/...'
      },
    ]
  }
});
```

After that I run first "**npm run build**" and "**npm run develop**" afterwards. I also don't see the plugin in the list of available plugins etc. I guess it must be something stupid - but I am unable to identify that issue.

**Comment by sarhugo** [comment]
Please check if adding the enabled and resolve attributes at config solves the issue. I just updated README. Config should look something like this:

```javascript
module.exports = ({ env }) => ({
  'cloudflare-pages': {
    enabled: true,
    resolve: "./src/plugins/cloudflare-pages",
    instances: [
      {
        name: "Production build",
        hook_url: 'https://api.cloudflare.com/client/v4/pages/webhooks/deploy_hooks/...'
      },
      {
        name: "Staging build",
        hook_url: 'https://api.cloudflare.com/client/v4/pages/webhooks/deploy_hooks/...'
      },
    ]
  }
});
```

**Comment by adam-karnet** [comment]
Thanks for you replies - now when I add those options, this is what happens on **npm run build** - 
```
Error: ./src/plugins/cloudflare-pages couldn't be resolved
    at toDetailedDeclaration (/Users/adam.karnet/Sites/localhost/admin-allwynentertainment-com/node_modules/@strapi/strapi/lib/core/loaders/plugins/get-enabled-plugins.js:38:15)
...
```
... and when I comment the **resolve** option out (so just **enabled** and **instances** remains) I end up with this error - 
```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
```
... which may be caused by the absence of resolve value, I get that. But I don't know what I am doing wrong so my behaviour differs from yours. Maybe we have different Node and Npm version? We use **Node 14.18.0** and **Npm 6.14.16** as Strapi suggests.

**Comment by adam-karnet** [comment]
After the last comment I updated to **Node 14.20.0** and **Npm 6.14.17** - and completely reinstalled all packages etc. but still ended up with the same behaviour that I described above.

**Comment by sarhugo** [comment]
The resolve property should point to the folder where the plugin was installed. Please check where it was installed (usually src/plugins)

**Comment by adam-karnet** [comment]
Maybe this would be a stupid question - but the plugin ends up in node_modules, doesn't it? Nothing gets created in ./src folder (and maybe that is the issue).
Strapi docs say that _Strapi automatically loads plugins installed with npm._ ... is there any other step that needs to be completed, so there are some files for the plugin in folder that you mentioned?

**Comment by adam-karnet** [comment]
**Update** - i checked your package.json and there is one thing missing which strapi documentation mentions as integral, it is key "kind" and it's value "plugin". When I update it in node_modules just for test, plugin gets recognised by the application.
```
"strapi": {
    "name": "cloudflare-pages",
    "displayName": "Cloudflare Pages",
    "description": "Publish changes via Cloudflare Pages.",
    "kind": "plugin"
  },
```
I updated it based on this info - https://strapi.io/blog/how-to-create-a-strapi-v4-plugin-publish-on-npm-6-6

After this change I can see the plugin in the application! But there are no instances now - plugin page says "_PLEASE CHOOSE WHICH PROJECT INSTANCE DO YOU WANT TO DEPLOY_", but there are no instances listed even when they are in the config as stated in docs. So that is a thing I am going to check next.

----------

**Update #2** - you also need to put your plugin data in config key in config/plugins.js like this - 
```
'cloudflare-pages': {
    enabled: true,
    config: {
      instances: [
        {
          name: "Production build",
          hook_url: 'https://api.cloudflare.com/client/v4/pages/webhooks/deploy_hooks/...'
        },
        {
          name: "Staging build",
          hook_url: 'https://api.cloudflare.com/client/v4/pages/webhooks/deploy_hooks/...'
        },
      ]
    }
  }
```
With the previous changes this is actually a working combination of sort. I can see instances listed and when I click the "Publish" button, modal element is displayed ... after confirmation deployment is triggered correctly. After that spinner gets displayed for all the instances (which is a bit confusing) though (we have production and staging environment) for few seconds. But I can create a different issue for that if you agree and want to solve it.

----------

Do you think you would be able to provide the **minor change in package.json** of the plugin, so it gets recognised by Strapi automatically?

**Comment by sarhugo** [comment]
Hi @adam-karnet , thanks for the research! later today I'll update package.json and also README to reflect the changes

**Comment by sarhugo** [comment]
Fixed on v0.4.1. Once again thanks @adam-karnet!
